src/ui/controls/edit.cpp

```diff
@@ -1256,14 +1260,25 @@ void CEdit::DrawColor(Math::Point pos, Math::Point dim, Gfx::Color color)
 // Give the text to edit.
 
 void CEdit::SetText(const char *text, bool bNew)
-{
+{
     int     i, j, font;
     bool    bBOL;
+
+    if ( text == nullptr )
+    {
+        return;
+    }
 
     if ( !bNew )  UndoMemorize(OPERUNDO_SPEC);
 
     m_len = strlen(text);
```

Comment:
Try convert all C style string functions to std::string equivalents whenever possible (in this case, also change function argument to std::string)